packages/react/src/components/DataTable/TableSelectAll.tsx

```diff
@@ -10,12 +10,20 @@ import React from 'react';
 import InlineCheckbox from '../InlineCheckbox';
 import cx from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
+import deprecate from '../../prop-types/deprecate';
 
 interface TableSelectAllProps {
   /**
    * Specify the aria label for the underlying input control
+   * node
    */
-  ariaLabel: string;
+  ['aria-label']?: string;
+
+  /**
+   * Deprecated, please use `aria-label` instead.
```

Comment:
There's actually a formal [deprecated tag in JSDoc](https://jsdoc.app/tags-deprecated) we could use here
```suggestion
   * @deprecated please use `aria-label` instead.
```

Reply:
Sorry, I didn't received emails for this and forgot to monitor the PR, so I missed the suggestions. Thanks for the review and for taking care of the missing parts.